File: MuayeneYonetimPortali/MuayeneYonetimPortali.Web/Modules/ServerTypes/Appointments/AppointmentsForm.ts
import { ServiceLookupEditor, LookupEditor, DateEditor, PrefixedContext, initFormType } from "@serenity-is/corelib";

export interface AppointmentsForm {
    CityId: ServiceLookupEditor;
    DistrictId: LookupEditor;
    HospitalId: ServiceLookupEditor;
    DepartmentId: LookupEditor;
    DoctorId: ServiceLookupEditor;
    PatientId: ServiceLookupEditor;
    AppointmentDate: DateEditor;
}

export class AppointmentsForm extends PrefixedContext {
    static readonly formKey = 'Appointments.Appointments';
    private static init: boolean;

    constructor(prefix: string) {
        super(prefix);

        if (!AppointmentsForm.init)  {
            AppointmentsForm.init = true;

            var w0 = ServiceLookupEditor;
            var w1 = LookupEditor;
            var w2 = DateEditor;

            initFormType(AppointmentsForm, [
                'CityId', w0,
                'DistrictId', w1,
                'HospitalId', w0,
                'DepartmentId', w1,
                'DoctorId', w0,
                'PatientId', w0,
                'AppointmentDate', w2
            ]);
        }
    }
}

File: MuayeneYonetimPortali/MuayeneYonetimPortali.Web/Modules/Appointments/Appointments/AppointmentsForm.cs
using Serenity.ComponentModel;
using System;

namespace MuayeneYonetimPortali.Appointments.Forms;

[FormScript("Appointments.Appointments")]
[BasedOnRow(typeof(AppointmentsRow), CheckNames = true)]
public class AppointmentsForm
{
    [Tab("Muayene Yeri")]
    public int CityId { get; set; }
    public int DistrictId { get; set; }
    public int HospitalId { get; set; }
    public int DepartmentId { get; set; }
    public int DoctorId { get; set; }
    public int PatientId { get; set; }

    [Tab("Muayene Tarihi")]
    [Category("Tarih Seç")]
    [DateTimeEditor(IntervalMinutes =20), HalfWidth]
    public DateTime AppointmentDate { get; set; }
}